app/gui2/shared/languageServer.ts

```diff
@@ -159,7 +159,12 @@ export class LanguageServer extends ObservableV2<Notifications & TransportEvents
       }
     })
     const reinitializeCb = () => {
-      if (!this.shouldReconnect) return
+      if (!this.shouldReconnect) {
```

Comment:
Please rename `reinitializeCb` to something like `onClosedCb` - it's no longer only about initialization.